istioctl/cmd/waypoint.go

```diff
@@ -115,6 +119,38 @@ func waypointCmd() *cobra.Command {
 			return nil
 		},
 	}
+	waypointDeleteCmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete a waypoint configuration",
+		Long:  "Delete a waypoint configuration from the cluster",
+		Example: `  # Delete a waypoint from the current namespace
+  istioctl x waypoint delete
+  
+  # Delete a waypoint from a specific namespace for a specific service account
+  istioctl x waypoint delete --service-account something --namespace default`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			gw := makeGateway(true)
```

Comment:
Why do you need this if you set it again few lines later?

Reply:
It is used to get the gw name with the same logic like `apply/generate` easily. The get function is just to ensure it is there and print the log if it is not found. I think it is redundent, and I'll remove it.